### PR TITLE
clang-tidy: rework Makefile and CI

### PR DESCRIPTION
**Pull request by carlopi**
#### Developer side
Add TIDY_CHECKS option, can be used like:
```
TIDY_CHECKS='cppcoreguidelines-narrowing-conversions' make tidy_check
```

Those options are provided on top of the standard list in `.clang-tidy` file.

Examples:
`'cppcoreguidelines-narrowing-conversions'` = specified test, plus all default ones 
`'*'` = all
`'-*,performance-*,bugprone-*'` = all performance or bugprone, none of the rest (= no default ones)

One thing to check is that via `pip install clang_tidy` I upgraded the version I had, and I while current options work with the most recent one (e.g. `-*` has to be explicitly specified to remove recommended checks), I have not checked this might impact current local workflows.
I think @Mytherin uses this for sure.
Note that empty `TIDY_CHECK` keeps current semantic, so it should be OK.

#### CI side
Bumped ubuntu from 22 to 24, that in turns means a newer clang-tidy. Also specified a subset of checks that are currently not failing, taken from the list of previously run checks, minus `readability-identifier-naming` that now has been improved and trigger in 2 instances (weak_ptr and enable_from_this class names).

Brief math on the list-checks output reveals there are around 500 (some might be aliases) and we are failing at least 100 unique categories. Unsure which one are relevant to tackle, I would consider `cppcoreguidelines-narrowing-conversions` as one of the first to investigate.

There are probably plenty more that can and should be added at the current config.

One thing I am not sure: whether the list of currently checked / supported list of checks should live. One one end it would be simpler to edit as an extra file with the list of checks, then compiled to the comma separated list. On the other this is simpler. I can look into moving that to a file, but can likely also be done as a follow up.

This PR also adds possibility to trigger the clang-tidy on a given subset of checks, passed via workflow_dispatch input. Idea is that it could be handy to expand the list.

### CODE REVIEW

**Comment by carlopi** [comment]
Here an example of a CI run, on my fork, triggered with a custom set of checks (that are then failing): https://github.com/carlopi/duckdb/actions/runs/10029689111/job/27718408533

**Comment by carlopi** [comment]
Failure is not connected, and found also on nightly.

**Review by Mytherin — COMMENTED** [review]
Thanks! Adding the option looks good to me. I'm not sure if providing the entire set of checks is the best interface here - maybe we can just provide *extra* checks instead?

File: .github/workflows/CodeQuality.yml [hunk 1]
```diff
@@ -94,14 +97,15 @@ jobs:
...
...bugprone-virtual-near-miss,cppcoreguidelines-avoid-non-const-global-variables,cppcoreguidelines-interfaces-global-init,cppcoreguidelines-pro-type-const-cast,cppcoreguidelines-pro-type-cstyle-cast,cppcoreguidelines-slicing,cppcoreguidelines-virtual-class-destructor,google-build-using-namespace,google-explicit-constructor,google-global-names-in-headers,google-runtime-int,hicpp-exception-baseclass,misc-definitions-in-headers,misc-throw-by-value-catch-by-reference,modernize-use-bool-literals,modernize-use-emplace,modernize-use-nullptr,modernize-use-override,performance-faster-string-find,performance-for-range-copy,performance-implicit-conversion-in-loop,performance-inefficient-algorithm,performance-inefficient-vector-operation,performance-move-const-arg,performance-move-constructor-init,performance-no-automatic-move,performance-noexcept-move-constructor,performance-trivially-destructible,performance-type-promotion-in-math-fn,performance-unnecessary-copy-initialization,performance-unnecessary-value-param,readability-braces-around-statements,readability-container-size-empty,readability-identifier-naming,-readability-identifier-naming,cppcoreguidelines-rvalue-reference-param-not-moved' }}
```
[the remark this answers is not in the document]
Comment (carlopi) [reply .github/workflows/CodeQuality.yml:108]: I have to say I had not seen `.clang-tidy` file, was wondering where the magic came from, I was under the assumption those were the default set of tests.

Thanks!

**Comment by carlopi** [comment]
Minor changes to `.clang-tidy`, now setting is to specify additional checks or, via `-*,check1-name,check2-name` to first exclude default ones and then add the provided patterns.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**